### DISCUSSION TITLE
Lesson 4 Practice

### DISCUSSION THREAD

**Discussion by yeahChibyke** [Cyfrin/foundry-full-course-cu#651] [Q&A] [answered]
I finished the first video and I am building a lot of smart contracts to solidify my knowledge on what I have learnt so far.
I want to write a hybrid smart contract for buying AAPL with USD. I am using Chainlink for the price feed, but there is no price feed available for Sepolia Testnet.
I copied the interface from Chainlink's Github repo, but I have some questions.
1) Does one aggregatorv3interface work for all Chainlink price feeds? (I used the AAPL/USD address)
2) I have attached the gist link of what I have written so far, but I have no idea if it works as I do not have any ETH to test it. (I basically wrote functions for getVersion() and getDescription()) Can anyone please help check if this works, even if it just by looking at it?

 https://gist.github.com/yeahChibyke/7d50b77324e9975422dd648598a8b22f

**Comment by theirrationalone** [comment]
Answer 1. Yes of course, goto `doc.chain.link` -> `pricefeed addresses` -> `sepolia` all mentioned pricefeeds will work. 

Answer2. You should you import instead.... 

```solidity
import "@chainlink/contracts/src/v0.8/interfaces/AggregatorV3Interface.sol"
```

I've seen your code and sorry to say, but that will not work that's and independent code no way to pull offchain data from oracle chainlink that's why you should use import like i mentioned above.

**Reply by yeahChibyke** [reply]
Answer 1: Ok. Thanks for clarifying that for me. 
Answer 2: I know my code is very incomplete :) I just want to know if I call any function in the aggregator interface, will it work? I had no way to test it out since I don't have ETH. And, there is AAPL/USD price feed for Sepolia on Chainlink.
Thanks for reminding me concerning imports. I will do my best not to forget :)

**Reply by theirrationalone** [reply]
Need more Sepolia ETH: https://sepoliafaucet.com/

`Login|Signup` required!

**Comment by yeahChibyke** [comment]
The AAPL/USD data feed is only available on Ethereum Mainnet. My question is: 
Is it  the same aggregator for both Testnet and Mainnet?

**Reply by theirrationalone** [reply]
Obviously Yesss !!

**Reply by yeahChibyke** [reply, marked as answer]
Ok. Thank you very much!!!
